src/ChunkedBase.jl

```diff
@@ -114,6 +50,14 @@ export SkipContext
 export Lexer
 export parse_file_serial, parse_file_parallel, populate_result_buffer!
 
+
+# By uncommenting all `# TRACING #` in the package, e.g. by turning them to `#= ... =#`,
+# you'll enable low-overhead tracing capability.
```

Comment:
Yeah, related to your TODO below, we should talk about how you could change this to be behind a compile-time debug flag like we do in raicode! Something like this:
```julia
@inline TRACING() = false

function enable_tracing(enabled::Bool)
    # Only update the method when the level has actually changed, otherwise we
    # get unnecessary compilation churn from the change in the method table
    if enabled != Base.invokelatest(TRACING)
        @eval @inline $(@__MODULE__).TRACING() = $enabled
    end
end
```
Then you can put your tracing statements behind a flag like
```
if TRACING
...
end
```
inside all your code, or use a macro that inserts that.

And IMO you can define all of the code below in a debugging module and if it's never used it won't bother anything, so no need to have it commented out. Just put it in a separate file.

Reply:
^ But you don't need to do any of this as part of this project

Reply:
Thanks!

Reply:
please open an issue about this if there isn't one